app/src/components/common/SharingModal/Workspaces/ShareFromPrivate.tsx

```diff
@@ -45,7 +45,11 @@ export const ShareFromPrivate: React.FC<Props> = ({ selectedRules, setPostShareV
       user?.details?.profile?.uid
     );
     try {
-      const teamName = isBusinessUser ? getDomainFromEmail(user?.details?.profile?.email) : "My Team";
+      const teamName = isBusinessUser
+        ? `${user.details?.profile?.displayName?.split(" ")[0]}'s team (${
+            getDomainFromEmail(user?.details?.profile?.email).split(".")[0]
```

Comment:
Have a common util function for this, also `user.details?.profile?.displayName` have a null check for `user` as it can be undefined.